src/codeu/chat/database/Database.java

```diff
@@ -0,0 +1,55 @@
+package codeu.chat.database;
+
+import java.sql.*;
+
+import codeu.chat.util.Logger;
+
+public final class Database {
+
+  private final static Logger.Log LOG = Logger.newLog(Database.class);
+
+  private String m_dbPath;
+  private Connection m_connection;
+
+  public Database(String dbPath) {
+      m_dbPath = dbPath;
+  }
+
+  /**
+   * Attempts to validate the connection.
+   * If the connection is closed or does not exist, the function will try
+   * to reconnect.
+   *
+   * @return Whether the connection was succesfully validated.
+   */
+  private boolean validateConnection() {
+    try {
+      // If the connection is null or is closed, create a new one.
+      if (m_connection == null || !m_connection.isValid(0)) {
+        Class.forName("org.sqlite.JDBC");
+        m_connection = DriverManager.getConnection(String.format("jdbc:sqlite:%s", m_dbPath));
+      }
+
+      // Any other state implies the connection is valid.
+      return true;
+    } catch (Exception ex) {
```

Comment:
In this particular case, I feel this is a non-recoverable error. There is nothing the server will be able to do if the database connection fails.

Since this is sqlite, I would expect that the connection will succeed as long as the database path is valid.

I would rather terminate this program right away when this exception occurs, instead of returning false. Otherwise, you will end up building more levels of checking on top of this when your system is already in a bad and non-recoverable state.

You could possibly add more information to STDOUT advising the user to check the database path, etc. and pass the right set of arguments.